bundles/org.openhab.binding.androidtv/README.md

```diff
@@ -0,0 +1,484 @@
+Dimmer ShieldTV_VOLUME "VOLUME [%s]" { channel = "androidtv:shieldtv:livingroom:volume" }
+Switch ShieldTV_MUTE "MUTE [%s]" { channel = "androidtv:shieldtv:livingroom:mute" }
+
+String GoogleTV_KEYBOARD "KEYBOARD [%s]" { channel = "androidtv:googletv:theater:keyboard" }
+String GoogleTV_KEYPRESS "KEYPRESS [%s]" { channel = "androidtv:googletv:theater:keypress" }
+String GoogleTV_KEYCODE "KEYCODE [%s]" { channel = "androidtv:googletv:theater:keycode" }
+String GoogleTV_PINCODE  "PINCODE [%s]" { channel = "androidtv:googletv:theater:pincode" }
+String GoogleTV_APP "APP [%s]" { channel = "androidtv:googletv:theater:app" }
+Player GoogleTV_PLAYER "PLAYER [%s]" { channel = "androidtv:googletv:theater:player" }
+Switch GoogleTV_POWER "POWER [%s]" { channel = "androidtv:googletv:theater:power" }
+Dimmer GoogleTV_VOLUME "VOLUME [%s]" { channel = "androidtv:googletv:theater:volume" }
+Switch GoogleTV_MUTE "MUTE [%s]" { channel = "androidtv:googletv:theater:mute" }
+```
+
+KEYPRESS will accept the following commands as strings (case sensitive):
+
+- KEY_UP
+- KEY_DOWN
+- KEY_RIGHT
+- KEY_LEFT
+- KEY_ENTER
+- KEY_HOME
+- KEY_BACK
+- KEY_MENU
+- KEY_PLAY
+- KEY_PAUSE
+- KEY_PLAYPAUSE
+- KEY_STOP
+- KEY_NEXT
+- KEY_PREVIOUS
+- KEY_REWIND
+- KEY_FORWARD
+- KEY_POWER
+- KEY_GOOGLE
+- KEY_VOLUP
+- KEY_VOLDOWN
+- KEY_MUTE
+- KEY_SUBMIT
+
+The list above causes an instantanious "press and release" of each button.  
+If you would like to manually control the press and release of each you may append _PRESS and _RELEASE to the end of each.
+(e.g. KEY_FORWARD_PRESS or KEY_FORWARD_RELEASE)
+
+You may also send an ASCII character as a single letter to simulate a key entry (e.g KEY_A, KEY_1, KEY_z).
+Use KEY_SUBMIT when full text entry is complete to tell the shield to process the line.
+KEY_SUBMIT is automatically sent by KEYBOARD when a command is sent to the channel.
+
+APP will display the currently active app as presented by the AndroidTV.  
+You may also send it a command of the app package name (e.g. com.google.android.youtube.tv) to start/change-to that app.
+
+KEYCODE values are listed at the bottom of this README.
+NOTE: Not all KEYCODES work on all devices.  Keycodes above 255 have not been tested.
+
+## Pin Code Process
+
+For the AndroidTV to be successfully accessed an on-screen PIN authentication is required on the first connection.  
+
+To begin the PIN process, send the text "REQUEST" to the pincode channel while watching your AndroidTV.
+
+A 6 digit PIN should be displayed on the screen.
+
+To complete the PIN process, send the PIN displayed to the pincode channel.
+
+The display should return back to where it was originally.
+
+If you are on a ShieldTV you must run that process a second time to authenticate the GoogleTV protocol stack.
+
+This completes the PIN process.
+
+Upon reconnection (either from reconfiguration or a restart of OpenHAB), you should now see a message of "Login Successful" in openhab.log
```

Comment:
Just a quick comment for this process: Consider adding a console command for providing the PIN. It seems a bit cumbersome to temporarily create an item in order link it to a channel, just to delete it again afterwards.

I admit it might be a matter of preference. Unfortunately I can't think of other similar processes right now and how they are implemented, except perhaps OAuth2 flow where servlets are often used.

Reply:
I'm not sure how to do that. I use the item/channel via karaf personally. E.g. "openhab:send androidtv_pin REQUEST".

Reply:
If already using the console, it would be more convenient with a custom command for this to avoid the need for a linked item. See for example:

https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/console/BoschIndegoCommandExtension.java

Documented as last step here:

https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.boschindego/README.md#authorization

Reply:
I'll give that a try. Standby.

Reply:
Is there any reason I couldn't just use the CommandExtension as a go-between to the handleCommand() already existing in the handler?  There is a good bit of logic there already to determine what to send to where on each of the connection managers.  It would be trivial if I could just send a "fake command" to handleCommand().

Reply:
I looked through what you posted.  Similar to the TranslationProvider, no issues adding this but I'd rather not make large changes to the PR at this point.  I like the idea, I have no problem saying I'll implement it in short order, but I'd rather do it as a small PR next.  After going back to look at the logic in the handler, I'd have to pull a few things out in the connection managers to create a pin function that I could call directly instead of having it buried in the handleCommand() logic.  This is going to need additional testing as well to determine that we don't have a regression across the different hardware platforms that this supports.